src/main/java/Task.java

```diff
@@ -0,0 +1,96 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * An <code>abstract</code> class representing a task which has name and status (whether it is done) fields.
+ */
+abstract public class Task {
+    private static final int CURRENT_YEAR = Calendar.getInstance().get(Calendar.YEAR);
+    private static SimpleDateFormat dateFormatter = new SimpleDateFormat("d/M HHmm");
```

Comment:
it would be better if dateFormatter was decoupled from Task.

consider having a DateUtil class.

Reply:
Didn't think of that. However, now that you mention it, I think that it is indeed a good point. Thanks!